app/models/apple/podcast_delivery.rb

```diff
@@ -2,74 +2,151 @@
 
 module Apple
   class PodcastDelivery < ActiveRecord::Base
+    include Apple::ApiResponse
+
     serialize :api_response, JSON
 
-    has_one :podcast_delivery_file
+    has_many :podcast_delivery_files
     belongs_to :episode, class_name: "::Episode"
     belongs_to :podcast_container, class_name: "::Apple::PodcastContainer"
 
+    delegate :apple_episode_id, to: :podcast_container
+
     enum status: {
       awaiting_upload: "AWAITING_UPLOAD",
       completed: "COMPLETED",
       failed: "FAILED"
     }
 
-    def self.create_podcast_deliveries(api, episodes_to_sync)
-      episodes_needing_delivery = episodes_to_sync.reject { |ep| ep.podcast_container.podcast_delivery.present? }
+    def self.update_podcast_deliveries_state(api, episodes)
+      results = get_podcast_deliveries_via_containers(api, episodes.map(&:podcast_container))
+
+      join_on_apple_episode_id(episodes, results).map do |(episode, delivery_row)|
+        upsert_podcast_delivery(episode, delivery_row)
```

Comment:
Same pattern as podcast_containers:

1) get resources via hypermedia graph
2) join results to the episodes here and upsert local state